### PR TITLE
split this package into -node and -wasm

### PR DESCRIPTION
**Pull request by staltz**
- https://github.com/ssb-ngi-pointer/ssb-validate2-rsjs-node
- https://github.com/ssb-ngi-pointer/ssb-validate2-rsjs-wasm

### Doesn't fully work, but should be merged

I think that ssb-validate2-rsjs after merging will not work out of the box, because `-node` would be missing automatic *build-when-npm-installed* functionality, so `-node` would be installed without being built, and thus wouldn't execute in runtime. But I think it's okay to merge this PR, because we're going to solve "build-when-npm-installed" in PR #19 which will make it work for mobile operating systems, but should cover also non-mobile operating systems. 

### About prebuilds...

Note, we can (and probably should) ship prebuilds, but we weren't doing that anyway before this PR. And shipping `dist/index.node` doesn't qualify as a prebuild because it would only support one operating system. What we need instead is a folder `prebuilds/$PLATFORM-$ABI/index.node` for combinations of `$PLATFORM` and `$ABI`, and if the appropriate prebuild is missing, we need to let `-node` do the "build-when-npm-installed" behavior.

### TL;DR

npm installing `ssb-validate2-rsjs` currently doesn't work anyway, and still doesn't work after merging this PR. Let's address that issue in other PRs.

### CODE REVIEW

**Review by mycognosist — APPROVED** [review]
Makes sense to me.

>  What we need instead is a folder prebuilds/$PLATFORM-$ABI/index.node for combinations of $PLATFORM and $ABI, and if the appropriate prebuild is missing, we need to let -node do the "build-when-npm-installed" behavior.

👍🏻
